torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -851,6 +851,28 @@ static void removeSequenceSplitConcat(Block* b) {
   }
 }
 
+static void insertIdentityForInputUsedAsOutput(Block* b) {
+  // Resolving limitation from ONNX that the block input cannot be used directly
+  // as block output. Inserting an Identity node inside
+  // the block, linking with the value as workaround.
+  for (auto out : b->outputs()) {
+    auto n = out->node();
+    if (nullptr != n && n->kind() == prim::Param) {
+      Node* id_node = b->owningGraph()->create(onnx::Identity);
+      id_node->insertBefore(b->return_node());
```

Comment:
we may have multiple outputs `b->outputs()` here, why in all the cases, `id_node->insertBefore(b->return_node());` is the same? I thought it would dependent on different `out` here.

Reply:
`param_x ->b_return_node` => `param_x -> identity_x -> b_return_node`